OpenInFirefoxClient/OpenInFirefoxControllerObjC.m

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#import <UIKit/UIKit.h>
+#import "OpenInFirefoxControllerObjC.h"
+
+static NSString *const firefoxScheme = @"firefox:";
+static NSString *const myAppScheme = @"firefoxopeninfirefoxclient://";
+
+@implementation OpenInFirefoxControllerObjC
+
+//Creates a shared instance of the controller.
++ (OpenInFirefoxControllerObjC *)sharedInstance {
+  static OpenInFirefoxControllerObjC *sharedInstance;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    sharedInstance = [[self alloc] init];
+  });
+  return sharedInstance;
+}
+
+//Custom function that does complete percent escape for constructing the URL
```

Comment:
Nit: period after comment, space after `//` (here and below).

Apologize for the OCD style, but since this is code that we plan to expose to devs, it should be well-edited.